doc/ROADMAP.rst.txt

```diff
@@ -0,0 +1,377 @@
+holes.  Additionally the coverage should be tracked over time and we should
+ensure it only goes up.
+
+Besides coverage there is also the issue of correctness - older code may have a
+few tests that provide decent statement coverage, but that doesn't necessarily
+say much about whether the code does what it says on the box.  Therefore code
+review of some parts of the code (``stats`` and ``signal`` in particular) is
+necessary.
+
+
+Documentation
+`````````````
+The documentation is in decent shape.  Expanding of current docstrings and
+putting them in the standard numpy format should continue, so the number of
+reST errors and glitches in the html docs decreases.  Most modules also have a
+tutorial in the reference guide that is a good introduction, however there are
+a few missing or incomplete tutorials - this should be fixed.
+
+
+Other
+`````
+Scipy 1.0 will likely contain more backwards-incompatible changes than a minor
+release.  Therefore we will have a longer-lived maintenance branch of the last
+0.X release.
+
+Regarding Cython code:
+
+  - It's not clear how much functionality can be Cythonized without making the
+    .so files too large.  This needs measuring.
+  - Cython's old syntax for using numpy arrays should be removed and replaced
+    with Cython memoryviews.
+  - New feature idea: more of the currently wrapped libraries should export
+    Cython-importable versions that can be used without linking.
+
+Regarding build environments:
+
+  - NumPy and SciPy should both build from source on Windows with a (possibly
+    modified) MinGW or MinGW-w64 toolchain and be compatible with Python
+    installations compiled with either the same MinGW or with MSVC.
+  - Bento will be officially supported as the second build tool besides
+    distutils.  At the moment it still has an experimental,
+    use-at-your-own-risk status, but that has to change.
+
+A more complete continuous integration setup is needed; at the moment we often
+find out right before a release that there are issues on some less-often used
+platform or Python version.  At least needed are Windows (MSVC and MinGW),
+Linux and OS X builds, coverage of the lowest and highest Python and Numpy
+versions that are supported, a Bento build and a PEP8/PyFlakes checker.
+
+Modules
+-------
+
+cluster
+```````
+This module is in good shape, after having been rewritten in Cython during a
+GSoC'14 project.
+
+
+constants
+`````````
+This module is basically done, low-maintenance and without open issues.
+
+
+fftpack
+```````
+Needed:
+
+  - solve issues with single precision: large errors, disabled for difficult sizes
+  - fix caching bug
+  - Bluestein algorithm nice to have, padding is alternative
+  - deprecate fftpack.convolve as public function (was not meant to be public),
+    resolve differences between ``signal.fftconvolve`` / ``fftpack.convolve`` /
+    ``signal.convolve`` and ``numpy.convolve`` 
```

Comment:
gh-5608 just added a new feature to ``signal.convolve`` right? IIRC there was a thing with normalization conventions or something similar. Can't find the details summarized anywhere though. I think it's a matter of someone sitting down and comparing those 3 functions. Probably doesn't need to be on this roadmap though, can open an issue for it.